src/modules/ekf2/ekf2_main.cpp

```diff
@@ -165,11 +198,21 @@ class Ekf2 final : public ModuleBase<Ekf2>, public ModuleParams
 	const float _vel_innov_spike_lim = 2.0f * _vel_innov_test_lim;	///< preflight velocity innovation spike limit (m/sec)
 	const float _hgt_innov_spike_lim = 2.0f * _hgt_innov_test_lim;	///< preflight position innovation spike limit (m)
 
+	// GPS blending and switching
+	struct gps_message _gps_state[GPS_MAX_RECEIVERS + 1]; ///< internal state data for the physical and blended GPS
+	struct gps_message _gps_output[GPS_MAX_RECEIVERS + 1]; ///< output state data for the physical and blended GPS
```

Comment:
Why is it necessary to store _gps_output? Couldn't it be discarded at the end of each update_gps_blend_states?

Reply:
We can't discard it until after GPS data has been written to the EKF buffer.

Reply:
I will refactor the code to remove the need for that class variable. Refactoring needs to happen anyway because there is a corner case that is not being handled correctly at the moment that results in the output not updating if one of the receivers loses lock. This was picked up in dynamic testing.

Reply:
I've split the calc_gps_blend_outputs into 3 functions for updating of receiver offsets, application of receiver offsets and calculation of blended  receiver output.

I'm using a class variable for outputs because I am unsure how to pass an array of structs by reference without creating a new type, and the compiler would not let me #include <array> and use std::array to create a new type